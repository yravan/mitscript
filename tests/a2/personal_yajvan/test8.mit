// Native Functions Test
print("\\n");
a = {x:1; y:"10";};
print(intcast(a.y) == 10);
oldprint = print;
print = fun(){
    oldprint("Level 1");
    oldprint(" ");
};

newprint = print;
print = fun(){
    oldprint("Level 2");
    newprint();
};

newnewprint = print;
print = fun(){
    oldprint("Level 3");
    newnewprint();
};
print();

//oldprint(input());