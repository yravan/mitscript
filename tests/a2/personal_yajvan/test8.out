\n
true
Level 3
Level 2
Level 1
 
